sdk/bare/common/drv/eddy_current_sensor.c

```diff
@@ -40,26 +42,39 @@ void eddy_current_sensor_set_divider(uint8_t divider)
     Xil_Out32(EDDY_CURRENT_SENSOR_BASE_ADDR + (2 * sizeof(uint32_t)), divider);
 }
 
+double eddy_current_sensor_bits_to_voltage(uint32_t data)
+{
+    bool is_negative = 0x20000 & data;
+
+    // Convert 2's compliment to positive data
+    if (is_negative) {
+        data = ~data;
+        data += 1;
+    }
+
+    // Convert data to voltage (+/-5V)
+    double resolution = 0.00003814697;              // 5V / 2^17
+    double voltage = (0x1FFFF & data) * resolution; // 17-bit data
+
+    if (is_negative) {
+        return -voltage;
```

Comment:
Change to gain of -1, not return from `if` statement. Prefer single return at end of function.

```C
voltage = -voltage;
```